Add devise

lmunozm1702 opened the pull request:
## In this PR, @tamana-Balkhi  and I, implements:

### Remove current_user method in ApplicationController. Install and set up the devise gem.

### Can register a new user.
![image](https://user-images.githubusercontent.com/53025062/223879186-cf51c749-9e04-4773-9424-5178cb97d93d.png)

![image](https://user-images.githubusercontent.com/53025062/223879330-8633f47b-ea0e-49f2-ba92-72269a942135.png)

### User logs in with a combination of email and password.

![image](https://user-images.githubusercontent.com/53025062/224010832-59220808-1fa3-49bf-b50a-12ebff234c1a.png)

![image](https://user-images.githubusercontent.com/53025062/224010975-cb6444ed-d428-461b-aedb-d307ee6fe17e.png)

### Hashed password should be stored in the database.

![image](https://user-images.githubusercontent.com/53025062/223855430-093ddb89-8f96-4bef-b260-f258ecad5456.png)

### Ask for confirmation of email.
> Wrong email:
![image](https://user-images.githubusercontent.com/53025062/224075439-e045e9a9-9b5c-4108-a96a-e529f115074d.png)

![image](https://user-images.githubusercontent.com/53025062/224077710-a9229cf9-7d0d-4577-aaa9-913a0c988003.png)

> Correct email:
![image](https://user-images.githubusercontent.com/53025062/224077885-cd5eed1e-d6ca-45c4-9125-2709d3c8c15c.png)

![image](https://user-images.githubusercontent.com/53025062/224078056-a03ed84e-c46d-4a99-92ac-64494c2ce339.png)

### Can reset password.
![image](https://user-images.githubusercontent.com/53025062/224060288-61f48f81-59f8-4e89-b8f6-d677d350287b.png)

![image](https://user-images.githubusercontent.com/53025062/224060022-c612aa60-f815-40f1-9d51-f96955054830.png)

![image](https://user-images.githubusercontent.com/53025062/224059912-68d23df4-ed09-42df-8589-8057ddf6a336.png)

![image](https://user-images.githubusercontent.com/53025062/224059829-1a16d087-3f98-4d0c-a5f4-0cec4b0c0b74.png)

lmunozm1702 committed:
Install and configure DEVISE gem.

lmunozm1702 committed:
Flash messages styling

lmunozm1702 committed:
Create sign-up process

lmunozm1702 committed:
Add login with email and password and logout.

lmunozm1702 committed:
Style User list

Set default avatar for user.
Style logout button.

lmunozm1702 committed:
Style sign-up form

lmunozm1702 committed:
Style login page

lmunozm1702 committed:
add devise default views

lmunozm1702 committed:
Add email confirmation field in sign up process.

tamana-Balkhi committed:
make password editable for user

tamana-Balkhi committed:
create new password file

tamana-Balkhi committed:
fix rubucop offence

AshakaE commented on app/models/user.rb:
```diff
@@ -1,10 +1,17 @@
+
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
- [x] Kindly enable `:confirmable` for devise

lmunozm1702 replied:
Done!.
- Email sent to confirmate the new account
![image](https://user-images.githubusercontent.com/53025062/224111129-2c1191cc-e80a-457f-9966-794636045f3e.png)

- New account confirmed:
![image](https://user-images.githubusercontent.com/53025062/224111358-e41e1f66-1fc0-44f9-bdd3-1efd2bb52ed3.png)

- Activate and style for "Resend confirmation instructions"
![image](https://user-images.githubusercontent.com/53025062/224112699-06992c00-40e6-4603-8b4e-1710ad6618c1.png)

![image](https://user-images.githubusercontent.com/53025062/224112856-75b23e55-4b3c-4996-9c34-d981df553688.png)

![image](https://user-images.githubusercontent.com/53025062/224113164-36487014-b4d6-4cbb-815c-ded70fab5e39.png)

AshakaE reviewed (changes requested):
Hi @lmunozm1702  & @tamana-Balkhi  , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- So far so good, proper Git flow used
- I appreciate the screenshots 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

lmunozm1702 committed:
Fix Code Reviewer Issue

- Activate confirmable on /app/model/user.rb
- Style and test email confirmation flow and Resend confirmation instructions.

fobadara reviewed (approved):
Hi team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Linters are passing

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
